server/routes/api/section.js

```diff
@@ -159,4 +165,45 @@ router.get('/updateSection', (req, res) => {
     });
 });
 
-module.exports = router;
\ No newline at end of file
+module.exports = router;
+
+// ------------- Combined Seating Layout and Course Methods -------------
+
+// gets information regarding a course given a student/prof id
+router.get('/getCourseView', (req, res) => {
+    // this is the professor ID
+    let courseID = req.body.courseID;
+
+    Course.findOne( {_id: courseID} , function(err, course){
+        if(err){
+            console.log(err);
+            return res.status(500).send();
+        }
+
+        seatingLayout = SeatingLayout.findOne({_id: course.seating_layout}, function(err, seatingLayout){
+
+            let myLayout;
+            if(err){
+                console.log(err);
+                return res.status(500).send();
+            }
+    
+            if(seatingLayout != null && seatingLayout.layout != null) {
+                myLayout = seatingLayout.layout;
+                console.log("?");
+            } else {
+                myLayout = "Layout not found";
+            }
```

Comment:
I would suggest setting the layout to an empty array rather than a message since that will make it easier for the frontend (we won't have to check whether it is a string or an array, we can just loop through the array as normal but it will do nothing since it will be empty). 
Also we can remove the console log (unless you still need it?).
```suggestion
            if(seatingLayout != null && seatingLayout.layout != null) {
                myLayout = seatingLayout.layout;
            } else {
                myLayout = [];
            }
```